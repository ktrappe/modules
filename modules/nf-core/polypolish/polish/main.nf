process POLYPOLISH {
    tag "$meta.id"
    label 'process_high_memory'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/polypolish:0.6.0--hdbdd923_0':
        'biocontainers/polypolish:0.6.0' }"

    input:
    // input:  polypolish polish [OPTIONS] <ASSEMBLY> [SAM]
    tuple val(meta), path(fasta)
    tuple val(meta), path(sam)

    output:
    // polypolish polish draft.fasta filtered_1.sam filtered_2.sam > polished.fasta
    tuple val(meta), path("*.fasta"), emit: fasta
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    polypolish \\
        polish \\
        $args \\
        ${fasta} \\
        $sam > ${prefix}_polished.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        polypolish: \$(polypolish polish --version |& sed '1!d ; s/polypolish polish //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_polished.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        polypolish: \$(polypolish polish --version |& sed '1!d ; s/polypolish polish //')
    END_VERSIONS
    """
}
