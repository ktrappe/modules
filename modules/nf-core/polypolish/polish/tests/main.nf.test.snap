{
    "sarscov2 - sam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_polished.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,18c318eb5fdfc2389f8dbb2186cf389c"
                ],
                "fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_polished.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,18c318eb5fdfc2389f8dbb2186cf389c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-24T15:12:15.777523"
    },
    "sarscov2 - sam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_polished.fasta:md5,48c6f75c9950164f9cbe119c470b7db0"
                    ]
                ],
                "1": [
                    "versions.yml:md5,18c318eb5fdfc2389f8dbb2186cf389c"
                ],
                "fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_polished.fasta:md5,48c6f75c9950164f9cbe119c470b7db0"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,18c318eb5fdfc2389f8dbb2186cf389c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-24T15:33:02.359162"
    }
}